### PR TITLE
Deduplicate RetentionLease.source

### PR DESCRIPTION
**Pull request by original-brownbear**
We only ever see the strings "ccr" or "peer recovery" here. A recent experiment with ccr and many follower clusters following one leader, revealed 100s of MB being wasted on duplicate "ccr" strings when fetching index stats -> lets fix this by simple deduplication.

cc @dliappis from last night :)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java [hunk 1]
```diff
@@ -92,14 +92,17 @@ public RetentionLease(final String id, final long retainingSequenceNumber, final
         if (timestamp < 0) {
             throw new IllegalArgumentException("retention lease timestamp [" + timestamp + "] out of range");
         }
-        Objects.requireNonNull(source);
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java:95]: Even though the following line would throw the same NPE, I think it is nice to keep this as explicit requirement (similar one to above requireNonNull for the id)

File: server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java [hunk 2]
```diff
@@ -92,14 +92,17 @@ public RetentionLease(final String id, final long retainingSequenceNumber, final
...
         }
         this.id = id;
         this.retainingSequenceNumber = retainingSequenceNumber;
         this.timestamp = timestamp;
-        this.source = source;
+        this.source = switch (source) {
+            case "ccr" -> "ccr";
+            case ReplicationTracker.PEER_RECOVERY_RETENTION_LEASE_SOURCE -> ReplicationTracker.PEER_RECOVERY_RETENTION_LEASE_SOURCE;
+            default -> source;
+        };
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/index/seqno/RetentionLease.java:105]: Do you mind leaving a comment in the code explaining the purpose?
It looks quiet surprising without `StringLiteralDeduplicator` and the explanation in this pr.

**Review by idegtiarenko — APPROVED** [review]
Left couple of comments, otherwise :+1:

**Comment by original-brownbear** [comment]
Thanks Ievgen!

**Pull request merged**